src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1021,6 +1035,33 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
                                 summary='Problem loading the helm existing user supplied values')
         raise CLIInternalError("Problem loading the helm existing user supplied values: " + str(e))
 
+    # Change registry_path to private registry if PCR is set in helm values
+    existing_user_values_flat = utils.flatten(existing_user_values)
+    helm_container_repository = existing_user_values_flat.get('systemDefaultValues.image.repository')
+    registry_username = None
+    registry_password = None
+    if helm_container_repository.split('.')[0] != "mcr":
+        registry_path = "{}/azurearck8s/azure-arc-k8sagents:{}".format(helm_container_repository, arc_agent_version)
```

Comment:
Which version will be taken here if arc_agent_version is not passed by the customer? Is there a way to know the latest agent version present on pcr?

Reply:
changed it to connected_cluster.agent_version